### ISSUE TITLE
FYI: Don't init a project named `Sample`

### ISSUE THREAD

**Issue by magbicaleman** [ptmt/react-native-macos#122]
Os: 10.12.1 Beta (16B2338c)

Initialized a project named `Sample` and had duplicate name conflicts, and app loaded empty.

**Comment by ptmt** [comment]
How it looks like exactly? I tried this: 

```
react-native-macos init Sample
open Sample/macos/Sample.xcodeproj
```
and get working project with latest (0.10.x) react-native-macos.

**Comment by magbicaleman** [comment]
I removed my personal paths, but this it right here;

- `node version: v6.8.0`
- `nvm version: 0.29.0`
- `react-native-macos-cli: 0.2.1`

> Failed to build DependencyGraph: @providesModule naming collision:
  Duplicate module name: Sample
  Paths: /react-native/Sample/package.json collides with /react-native/Sample/node_modules/react-native-macos/Libraries/Sample/package.json

>This error is caused by a @providesModule declaration with the same name across two different files.
Error: @providesModule naming collision:
  Duplicate module name: Sample
  Paths: /react-native/Sample/package.json collides with react-native/Sample/node_modules/react-native-macos/Libraries/Sample/package.json

>This error is caused by a @providesModule declaration with the same name across two different files.
    at HasteMap._updateHasteMap (/react-native/Sample/node_modules/react-native-macos/packager/react-packager/src/node-haste/DependencyGraph/HasteMap.js:155:13)
    at /react-native/Sample/node_modules/react-native-macos/packager/react-packager/src/node-haste/DependencyGraph/HasteMap.js:130:31
~
Process terminated. Press <enter> to close the window

**Comment by ptmt** [comment]
Do you have the same problem with https://github.com/facebook/react-native? And can you try to init project in different folder (not a `react-native`)

**Comment by sampurcell93** [comment]
This type of error is often encountered when you have installation problems with watchman, at least for vanilla react native. I'd check your npm version, and make sure flow watchman is installed right.

**Comment by andrerfneves** [comment]
Tried doing some debugging with the latest RN iOS and RN macOS.

Specs:
```
node: v7.4.0
react-native-macos-cli: 0.2.1
react-native: 2.0.1
```

`react-native-macos init Sample` creates a project, that builds successfully but then errors out on the Packager when loading bundle:

![screen shot 2017-02-06 at 12 12 35 pm](https://cloud.githubusercontent.com/assets/2574011/22657763/9e8dc8da-ec65-11e6-9759-f6eafbc1f00b.png)

However, the exact same behavior is seen in React Native iOS:

![screen shot 2017-02-06 at 12 21 09 pm](https://cloud.githubusercontent.com/assets/2574011/22658104/e8688ce6-ec66-11e6-9bd9-ef8d52a6e106.png)

I've tried `resetting-cache`, tried cleaning my flow and watchman watchers. As well as reinstalling all modules or pruning npm cache.

Thoughts? @ptmt maybe we should close this issue since it's been a while, and `Sample` shouldn't be a good application name. As well as the overlapping bug behavior with React Native.
